six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -418,6 +418,21 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,
             }
         }
 
+        // SIDD 1.1 needs to read LUT directly from NITF
+        if (currentInfo->getData()->getDataType() == DataType::DERIVED &&
+            currentInfo->getData()->getVersion() == "1.1.0")
+        {
+            nitf::LookupTable nitfLut =
+                    subheader.getBandInfo(0).getLookupTable();
+
+            //getDisplayLUT will not return NULL for 1.1
+            currentInfo->getData()->getDisplayLUT().reset(
```

Comment:
This is done.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Thursday, July 14, 2016 8:43 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Lut reposition (#86)


In six/modules/c++/six/source/NITFReadControl.cpp<https://github.com/ngageoint/six-library/pull/86#discussion_r70907256>:

> @@ -418,6 +418,21 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,

>              }

>          }

>

> +        // SIDD 1.1 needs to read LUT directly from NITF

> +        if (currentInfo->getData()->getDataType() == DataType::DERIVED &&

> +            currentInfo->getData()->getVersion() == "1.1.0")

> +        {

> +            nitf::LookupTable nitfLut =

> +                    subheader.getBandInfo(0).getLookupTable();

> +

> +            //getDisplayLUT will not return NULL for 1.1

> +            currentInfo->getData()->getDisplayLUT().reset(

So rather than relying/hoping that someone always does this and remembers to call interleaveTable(), what about just adding a six::LUT constructor that takes in a nitf::LookupTable directly and does the right thing for you?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ngageoint/six-library/pull/86/files/43ba641118333d4806e6e7eb2f8a5468dedf4aa5#r70907256>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYPxJL4nXTE3UjSwmziLza82oX10vks5qVtelgaJpZM4JJsQh>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.